src/Microsoft.Azure.SignalR/HubHost/ServerLifetimeManager.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.SignalR.HubHost
+{
+    class ServerLifetimeManager
+    {
+        private readonly IList<Func<Task>> _shutdownHooks = new List<Func<Task>>();
+
+        public ServerLifetimeManager(
+            IServiceProvider provider
+        )
+        {
+#if NETCOREAPP
+            var lifetime = provider.GetService<IHostApplicationLifetime>();
+#elif NETSTANDARD
+            var lifetime = provider.GetService<IApplicationLifetime>();
+#else
+            var lifetime = null;

```

Comment:
could be reduced, as our package supports either NETSTANDARD or NETCOREAPP only? use if/else is enough?

Reply:
I think pointing out which platform/framework we have supported explicitly and throw `Exception` in which we haven't supported is better?

Reply:
I think it's not necessary, it won't be able to compile for unsupported framework. You may check AspNetCore repo, which doesn't has the if/elseif/else check like this.

---
In reply to: [396879153](https://github.com/Azure/azure-signalr/pull/857#discussion_r396879153) [](ancestors = 396879153)